### DISCUSSION TITLE
VPS for Master Server List with OpenRVS

### DISCUSSION THREAD

**Discussion by Grunt3327** [OpenRVS-devs/OpenRVS#56] [Ideas]
Hey gentleman, I have a few questions on how to set up a VPS so I can get all the servers up and running with the OpenRVS1.5. The end goal is to have what Tony had with the ability to add/delete/edit servers and for everyone to see the servers. I see in the OpenRVS that you can add where the URL is being pulled from, which you can see if you open the OpenRVS. I just don't know if I need a crazy amount of CPU coverage, RAM, or storage. If it is reasonable, I will set this VPS up and we can get a running master server list up. Would this be ideal for working in conjunction with OpenRVS? Would love any feedback from you all. Thank you for your time.

**Comment by willroberts** [comment]
Hello,

There is an OpenRVS server running here: https://openrvs.org/servers (pardon the expired cert; fortunately the game doesn't use SSL so it still works).

Here is the software we use to manage the server list: https://github.com/willroberts/openrvs-registry. The CPU/memory utilization is near zero, so it runs fine on a free-tier 1-core instance in AWS etc.

OpenRVS clients automatically use this server list, and OpenRVS servers automatically register with it (so no server list management or maintenance is needed).

Let me know if you have additional questions!

**Reply by Grunt3327** [reply]
Will,

Thank you for the quick response. I have seen the OpenRVS server and it has a short list of the servers. Now with the OpenRVS server, is there a way to add/delete/edit servers to that list? If not, this is something that I am interested in trying to fix, and would be open to getting a PVS to use, so that we are able to add/delete/edit servers within the master server list (if that makes sense). I don't know much on how all this is set up, but I am trying to understand what is needed to make this happen. 

With the software used to manage the server list, how does that work? Is there anything special needed i.e. programs, software? Like I said, I am trying to learn this. I like this game and Ravenshield is my pastime to just have fun and BS with other gamers. I do appreciate the time and I look forward to your thoughts and opinions. Thank you.

**Reply by willroberts** [reply]
The software has some built-in healthchecking (using the RVS Beacon system) which will hide servers which are not currently accepting requests. You can see the full list (including "unhealthy" servers) here: https://openrvs.org/servers/all

If there is a server which is incorrectly being marked as unhealthy, let me know and I'll take a look! Also keep in mind that automatic server registration only began with OpenRVS v1.5, so older servers won't appear automatically. It is possible to add them, though, by sending a beacon request on their behalf.

The software itself is just a binary built with Go; there are some instructions in the repo if you want to try it out. That binary can run on any VPS.

**Reply by Grunt3327** [reply]
OK, I really do appreciate it. As I said, I have no experience with writing code or anything, so trying to understand this the best I can. I just want to help where I can. Keeping it as simple as possible would be my ideal choice. So, what I understand is that there is a way to add additional servers by sending a beacon request on their behalf. How does that work? I do have a list of IP's if that helps. However, will this also be useful in adding/deleting old servers off the master list when seen in the game?

I am very curious about this Go software and willing to try it out. Again, I am hopeful that creating a VPS with I would think a "new" OpenRVS would be needed once it has been modified correctly to show the proper servers and the ability to add/delete/modify, instead of what is currently happening. Sorry, I find all of this fascinating and interesting. Learning new stuff is always pretty cool. Jack of all Trades, master of none.

**Reply by willroberts** [reply]
If you can share the IPs of any OpenRVS servers not showing up in the list, as well as the version of OpenRVS they are running, I can take a look and see what's going on.  
  
The "beacon" protocol is what the game uses to communicate with servers, and we can use it to query active servers to get their current map, game mode, player list, etc., or just to confirm that a server is online and reachable.

There's also an "active servers" web app here: https://openrvs.org/live. If you join a server, you'll see yourself on this page, which helps confirm that the server list/registry and beacons are all working as expected.

**Reply by Grunt3327** [reply]
Will,

Yes, I was trying to figure out the adding servers through HTTP portion and I must have been doing something wrong. However looking on the (http://www.openrvs.org/servers) all of ALLR6 is on there and a few others. The servers I will provide with their IP's are as follow:
|-HkM-| 104.238.222.81 Adversarial TDM
IEF 121.127.40.41 Adversarial TDM
iD 104.238.220.182 Adversarial TDM
RavenLanClan 75.188.142.62 COOP TH
ObsoleteSuperstars 172.93.104.79:7877 COOP TH
SSHQ 65.75.210.194:8777 COOP TH
SSHQ 186.233.187.33: 8977 Adversarial TDM

All of these should be running on OpenRVS1.5, since they were visible (this is just my thinking) in the master list when everything was working before the old hosting website went bye bye and Tony went to a better place (supposedly, which is sad). I hope that this helps and if you need any other information, please let me know. Again, I would like to help where I can and this is why I wanted to start a VPS so that I could add/delete/edit new servers. Thank you for helping and explaining things. It is much appreciated.

**Reply by willroberts** [reply]
I've just tried to manually healthcheck the beacon port (game port + 1000) on the servers you listed, with varying results:

- 104.238.222.81:7777 -> "not a valid openrvs beacon"
- 121.127.40.41:7777 -> "not a valid openrvs beacon"
- 104.238.220.182:7777 -> "not a valid openrvs beacon"
- 75.188.142.62:7777 -> "not a valid openrvs beacon"
- 172.93.104.79:8877 -> I see 5 players playing the BurningLibrary map ✅ 
- 65.76.210.194:9777 -> I see an empty server on the OS_Training_Extreme map ✅ 
- 186.233.187.33:9977 -> No response from the server.

Here are two more things for you to check, so we can figure out why they aren't automatically appearing:

1. What is the value of `ServerBeaconPort` in `RavenShield.ini` for each server? It should always be set to the value of `Port` plus 1000, so if a server is using `Port=6777`, it should also be using `ServerBeaconPort=7777`.
2. What is the value of `RegistryServerIP` in `openrvs.ini` (`RegistryServerHost` in OpenRVS 1.6+)? The registry moved hosts last year, so some older builds of OpenRVS may still be using the old value. The correct value is `184.73.85.28`.

If these two things are set correctly, then adding, editing, or removing servers is 100% automatic, with no manual intervention needed:

- **Adding servers** happens when a game server is first started, and a "registration" beacon is automatically sent to the OpenRVS registry. If this operation fails (for example, when your server can't reach openrvs.org, or when the openrvs.org server can't reach your game server), then the server won't appear in the list.
- **Editing servers** happens when a game server is restarted. The registry keys servers on `<ip>:<port>` identifiers, so everything else such as name, game mode, player count, etc. is pulled automatically from the server directly.
- **Removing servers** happens when the game server becomes unreachable. It will automatically be hidden after failing some number of consecutive health checks.

I suspect something is going wrong with the communication between your servers and the openrvs-registry (based on the value of `RegistryServerIP`), or with the communication from openrvs-registry to your servers (via `ServerBeaconPort`). Double check those and let me know what you find. Alternatively, the latest version of OpenRVS (1.6+) will have the correct config values built-in as well.

**Reply by Grunt3327** [reply]
I have reached out to those servers and they are looking into it. The last one, 186.233.187.33:8977 is a dead server for SSHQ, The other servers are current and running. I have reached out to those personnel to have a look see into what their settings are at. Once, I receive updates from those servers I will let you know. Thanks again.

**Reply by Grunt3327** [reply]
How would you like me to send these ServerBeaconPort, RegistryServerIP's? Unsure if having all that data out in the air is good or not, just curious before I send some information. Thanks

**Reply by willroberts** [reply]
Both of those are public, non-sensitive information. Basically, the `ServerBeaconPort` needs to be exactly 1000 more than the `Port` in `RavenShield.ini`, and they need to be using `RegistryServerIP=184.73.85.28` in `openrvs.ini`. Once those are done, it will "just work".

**Reply by Grunt3327** [reply]
I see, so there is really nothing to be sent to you, it would be sent to the OpenRVS program. Then with that new data the OpenRVS program would shoot a message to the hosting website that has all of the server lists (like Tony's) to update, then shoot right back going through the program lastly to finish update, right?  I think that makes sense to me lol.

**Reply by willroberts** [reply]
That's right, everything is automatic when working correctly. I suspect the `ServerBeaconPort` / `RegistryServerIP` are off for a couple of the servers not showing up.

**Comment by Legacy-TacticalGamingInteractive** [comment]
A lot of those servers went offline at some point way before this stuff with Tony went dead. He used to regularly clean up the list from ones that have been dead for months without responding. Some are from like even over a year ago. They were down ever since. I suspect wrong IP's now or they actually shut their servers down. Unlikely that so many would change their IPs. But who knows.
